quarkus/runtime/src/main/java/org/keycloak/quarkus/KeycloakRecorder.java

```diff
@@ -106,57 +109,102 @@ public void configSessionFactory(
+                            configHelpText += " " + newProp;
+                        }
+                    } else if (!BUILD_TIME_PROPERTIES.keySet().stream()
+                            .anyMatch(new Predicate<String>() {
+                                @Override
+                                public boolean test(String s) {
+                                    return PropertyMappers.canonicalFormat(finalPropertyName)
+                                            .equalsIgnoreCase(PropertyMappers.canonicalFormat(s));
+                                }
+                            })) {
+                        String prop = "--" + PropertyMappers.toCLIFormat(propertyName).substring(3) + "=" + value.getValue();
+
+                        if (!configHelpText.contains(prop)) {
+                            LOGGER.warnf("New property [%s] set with value [%s] in [%s]. This property is not persisted into the server image.",
+                                    propertyName, value.getValue(), value.getConfigSourceName(), buildValue);
+                            configHelpText += " " + prop;
+                        }
                     }
                 }
-            } else if (configHelpText != null && rebuild && isRuntimeValue(value)) {
-                String prop = "--" + PropertyMappers.toCLIFormat(propertyName).substring(3) + "=" + value.getValue();
-
-                if (!configHelpText.contains(prop)) {
-                    LOGGER.infof("New property [%s] set with value [%s] in [%s]. This property is not persisted into the server image.",
-                            propertyName, value.getValue(), value.getConfigSourceName(), buildValue);
-                    configHelpText += " " + prop;
-                }
             }
         }
 
         if (configArgs != null && !configArgs.equals(configHelpText)) {
-            LOGGER.infof("Please, run the 'config' command if you want to configure the server image with the new property values:\n\t%s config %s", Environment.getCommand(), String.join(" ", configHelpText.split(",")));
+            LOGGER.warnf("Please, run the 'config' command if you want to persist the new configuration into the server image:\n\n\t%s config %s\n", Environment.getCommand(), String.join(" ", configHelpText.split(",")));
         }
     }
 
-    private boolean isRuntimeValue(ConfigValue value) {
-        String name = value.getName();
-        return value.getValue() != null && !PropertyMappers.isBuildTimeProperty(name)
-                && !"kc.version".equals(name) && !"kc.config.args".equals(
-                name) && !"kc.home.dir".equals(name);
+    private boolean shouldValidate(String name, boolean rebuild) {
+        return rebuild && name.contains(MicroProfileConfigProvider.NS_KEYCLOAK_PREFIX) 
+                && (!PropertyMappers.isBuildTimeProperty(name)
+                && !"kc.version".equals(name) 
```

Comment:
Instead of this hardcoded list, we can maybe have some collection of those commands? Also it may be nice to have some class like "QuarkusConstants", so that we can use something like "!QuarkusConstants.KC_CONFIG_FILE.equals(name)" instead of "!"kc.config.file".equals(name) ?

However I don't consider this as a blocker for this PR, but rather something for later :-)